glommio/src/channels/local_channel.rs

```diff
@@ -269,22 +294,26 @@ impl<T> LocalSender<T> {
     ///     drop(receiver);
     /// });
     /// ```
+    #[inline]
     pub fn len(&self) -> usize {
-        let state = self.channel.state.borrow();
-        state.channel.len()
+        self.channel.state.borrow().channel.len()
     }
 
     fn wait_for_room(&self, cx: &mut Context<'_>) -> Poll<()> {
-        if !self.channel.is_full() {
-            Poll::Ready(())
-        } else {
-            let mut state = self.channel.state.borrow_mut();
-            state
-                .send_waiters
-                .as_mut()
-                .unwrap()
-                .push_back(cx.waker().clone());
-            Poll::Pending
+        // NOTE: it is important that the borrow is dropped
+        // if Poll::Pending is returned
+        self.channel.state.borrow_mut().wait_for_room(cx)
+    }
+}
+
+fn wake_up_all(ws: &mut Option<VecDeque<Waker>>) {
+    if let Some(ref mut waiters) = ws {
+        // replace core::mem::replace with core::mem::take,
+        // as soon as we drop support for rustc < 1.40.0
+        // we assume here that wakes don't try to acquire a borrow on
+        // the channel.state
+        for w in core::mem::replace(waiters, VecDeque::new()) {
```

Comment:
Do we still support `rustc < 1.40.0`? If not, I'll change this part of the code to use `core::mem::take`.

Reply:
we do not.

Reply:
For the future, I don't consider glommio API stable yet, and I believe I put a list of things I want to see happen before that in the docs website.

Until we declare the project API stable, I see no reason to ever support any version other than the current.